### DISCUSSION TITLE
Install extension for user

### DISCUSSION THREAD

**Discussion by JanPalasek** [quarto-dev/quarto-cli#4251] [Q&A]
Is there a possibility to install an extension per-user and not per-project? I would expect something like:

```shell
quarto install extension EXTENSION_NAME --user
```

And then I wouldn't need to install it for every project separately.

**Comment by dragonstyle** [comment]
There is not- we currently only support installing into a folder or project. We do this largely to try to aide reproducibility - the project folder will ideally contain all the resources required to render the document. We don't currently have a plan to support global per user or per machine installation of extensions.

We currently have a command `quarto create` - I wonder if it would be useful to somehow extend this with the idea of user defined templates which include project skeletons (with extensions). That would mean you could define some abstract idea of a project with extensions and just do `quarto create project <my-cool-project>` to configure a directory with everything that is needed to get it up and running...

**Reply by andrewpbray** [reply]
This could be really handy. I've been building out what amounts to a course website for college classes and it involves some extensions. This would be a great way to encapsulate the key parts of the structure: _quarto.yml, _variables.yml, _extensions, template .qmds, file structure.

**Reply by dragonstyle** [reply]
As I'm thinking about this, a good proxy for the time being would be creating a repo and using the `quarto use template` command, which will more or less accomplish this.

`quarto use template dragonstyle/my-template`

We document this feature primarily as useful to build Journal formats, but what is really happening is that we are automating the placement of files from the repo in a folder (screening out specific known files but otherwise just moving the contents into place as is, including installed extensions, template files, and so on). Worth considering...

https://quarto.org/docs/journals/formats.html

**Reply by andrewpbray** [reply]
Ah, interesting. I didn't realize that's what was going on under the hood. I'll look into that.
